bpo-43075: Fix ReDoS in  request (#24391)
python/cpython: Misc/NEWS.d/next/Security/2021-01-31-05-28-14.bpo-43075.DoAXqO.rst

> Fix ReDoS in request

Reviewer: Please be more informative. If you are Python user reading changelog, what it would say to you? What request? What type of ReDoS? Does it affect your code if it does so and so, or you can ignore it?

> Fix Regular Expression Denial of Service (ReDoS) vulnerability in urllib.request.  The ReDoS-vulnerable regex has quadratic worst-case complexity and it allows cause a denial of service when identifying crafted invalid RFCs. This issue only affects consumers using urllib.request.AbstractBasicAuthHandler.

Reviewer 2: ```suggestion
Fix Regular Expression Denial of Service (ReDoS) vulnerability in :class:`urllib.request.AbstractBasicAuthHandler`.  The ReDoS-vulnerable regex has quadratic worst-case complexity and it allows cause a denial of service when identifying crafted invalid RFCs.
```

Please explain that the issue is on the client server, the attacker needs to control the HTTP server.

> Fix Regular Expression Denial of Service (ReDoS) vulnerability in :class:`urllib.request.AbstractBasicAuthHandler`.  The ReDoS-vulnerable regex has quadratic worst-case complexity and it allows cause a denial of service when identifying crafted invalid RFCs. This ReDoS issue is on the client server and needs remote attackers to control the HTTP server.

Reviewer 2: "This ReDoS issue is on the client server"

Do you mean "client side"?

Author: Yes, it's the “client side”. I wrote it based on your reviews:joy:, and I'll change it right away. 

> Please explain that the issue is on the client server, the attacker needs to control the HTTP server.
